SimTKcommon/tests/TestStateXml.cpp

```diff
@@ -25,6 +25,46 @@
 
 using namespace SimTK;
 
+/* This function converts the provided instance of `T` to Xml, deserializes
+it from Xml, ensures the deserialized instance is equal to the original
+instance, reserializes the deserialized instance back to Xml and ensures
+the original desrialization and the reserialization are equal.
+
+One might set `checkInMemoryEquality` to false if `value` contains NaNs. */
```

Comment:
Ah good point about inf.


> This seems awkward

Because we have to manually say not to check for equality?

> Could you add explicit isNaN() checks?

Could you clarify? For non-scalar types, I want to make sure that specific elements are NaN/not NaN, rather than just checking that there *is* a NaN somewhere in the matrix/vector. Are you suggesting replacing the manual argument with `if (x.isNaN()) { skip operator==() }`?